pinot-spi/src/main/java/org/apache/pinot/spi/stream/PartitionLagState.java

```diff
@@ -27,12 +27,20 @@ public class PartitionLagState {
   protected final static String NOT_CALCULATED = "NOT_CALCULATED";
 
   /**
-   * Defines how far away the current record's offset / pointer is from upstream latest record
+   * Defines how far behind the current record's offset / pointer is from upstream latest record
    * The distance is based on actual record count.
    */
   public String getRecordsLag() {
     return NOT_CALCULATED;
   }
 
-  // TODO: Define record availability lag ($latest_record_consumption_time - $latest_record_ingestion_time)
+  /**
+   * Defines how soon after record ingestion was the record consumed by Pinot. That is, the difference between the
+   * time the record was consumed and the time at which the record was ingested upstream.
+   *
+   * @return Lag value in milliseconds
+   */
+  public String getRecordAvailabilityLag() {
```

Comment:
I didn't want to call it "offset" lag as that term doesn't makes sense for anything other than kafka. in fact, `recordsLag` can be computed only for Kafka. Hence, I am using the same term as kafka's metric name for lag.

For every other system, we can only compute a time-based lag. and it indicates when the record becomes available for consumption. I was considering calling it `getRecordAvailabilityLagMs`, if that helps clarify the meaning?

Reply:
Maybe `getRecordsLagMs`?
Or do you want to track other kinds of lags in the future? Not sure if `availability lag` is a common term in streaming

Reply:
> Maybe getRecordsLagMs? 

This would make one think that this lag is computed relative to the timestamp of the latest record in upstream (ie. latest_upstream_record_timestamp minus current_consumed_record_timestamp).  This is the lag in terms of record time relative to upstream. This is actually a metric that kinesis provides and recommends applications to monitor - `millisBehindLatest` ([here](https://docs.aws.amazon.com/kinesis/latest/APIReference/API_GetRecords.html#API_GetRecords_ResponseSyntax)). 

Where as, what we are computing here is relative to consumption time and the time at which the record was ingested upstream. I think it should be treated differently than the above and hence, I used the term `recordAvailabilityLag` as it shows how soon the record was consumed after it arrived in the stream. 

> Not sure if availability lag is a common term in streaming

I thought it was. At least it was in the kafka ecosystem. 

I am fine renaming the metric in this PR to `recordLagMs` and then, actually add another metric called `millisBehindLatest` (similar to kinesis). Wdyt?

Reply:
another alternative:
rename `recordAvailabilityLagMs` -> `consumeLagMs` and use `recordTimeLagMs` for `millisBehindLatest` type of metric

Reply:
I see. `recordAvailabilityLagMs` and `recordTimeLagMs` sounds good to me.